File: picturepicker/src/main/java/com/github2136/picturepicker/adapter/PhotoAdapter.java
package com.github2136.picturepicker.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.github2136.picturepicker.fragment.PhotoFragment;

import java.util.List;

/**
 * Created by yubin on 2017/9/20.
 */

public class PhotoAdapter extends FragmentStatePagerAdapter {
    private List<String> mPhotoPath;

    public PhotoAdapter(FragmentManager fm, List<String> photoPath) {
        super(fm);
        mPhotoPath = photoPath;
    }

    @Override
    public Fragment getItem(int position) {
        return PhotoFragment.newInstance(mPhotoPath.get(position));
    }

    @Override
    public int getCount() {
        return mPhotoPath.size();
    }
}

File: photopicker/src/main/java/com/github2136/photopicker/adapter/PhotoAdapter.kt
package com.github2136.photopicker.adapter

import com.github2136.photopicker.fragment.PhotoFragment

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

/**
 * Created by yb on 2017/9/20.
 */

class PhotoAdapter(fm: FragmentManager, private val mPhotoPath: List<String>) : FragmentStatePagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return PhotoFragment.newInstance(mPhotoPath[position])
    }

    override fun getCount(): Int {
        return mPhotoPath.size
    }
}
